src/components/Article.jsx

```diff
@@ -1,3 +1,65 @@
+import { useParams } from "react-router-dom"
+import {getArticleById, patchArticleById} from '../utils/API'
+import { useState,useEffect } from "react"
+import Like from '../icons/thumb-up.png'
+import dislike from '../icons/dislike.png'
+import '../styles/Article.css'
+import { Link } from "react-router-dom"
+
 export const Article = () =>{
-    
+    const {article_id} = useParams()
+    const [article,setArticle] = useState({})
+    const [like , setLike] = useState(0)
+    const [undo, setUndo] = useState(0)
+
+    useEffect(()=>{
+        getArticleById(article_id).then(article=>{
+            setArticle(article)})
+    },[article_id])
+
+    useEffect(()=>{
+        patchArticleById(article_id,like)
+    },[like])
+
+    useEffect(()=>{
+        patchArticleById(article_id,undo)
+    },[undo])
+
+    const voteHandler = (e,value) =>{
+        setArticle({...article,votes:article.votes+value})
+        setLike(like+value)
+    }
+
+    const undoHandler = (e,value) =>{
+        setArticle({...article,votes:article.votes+value})
```

Comment:
In line with my comment above, this isn't very DRY - vote and undoHandler are reeeallly similar in their functionality. It would be better to have one function with some conditional logic that handles the setUndo part if necessary.